mne/time_frequency/tfr.py

```diff
@@ -337,8 +361,12 @@ def cwt(X, Ws, use_fft=True, mode='same', decim=1):
 def _time_frequency(X, Ws, use_fft, decim):
     """Aux of time_frequency for parallel computing over channels
     """
-    n_epochs, n_times = X.shape
-    n_times = n_times // decim + bool(n_times % decim)
+    if isinstance(decim, int):
+        decim = slice(None, None, decim)
+    elif not isinstance(decim, slice):
+        raise(TypeError, '`decim` must be int or slice, got %s instead'
+                         % type(decim))
```

Comment:
Since you're modifying this private function, can you update the docstring of its calling functions?

from what I see, there s at least `tfr_morlet`, `tfr_multitaper`, `_induced_power_mtm`, `_induced_power_cwt `.

You probably need to add tests for these too?